controller_manager:
  ros__parameters:
    update_rate: 100 # Hz
    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_controller:
      type: parallel_gripper_action_controller/GripperActionController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    gravity_compensation_controller:
      type: trossen_arm_controllers/GravityCompensationController

    cartesian_pose_controller:
      type: trossen_arm_controllers/CartesianPoseController

arm_controller:
  ros__parameters:
    joints:
      - joint_0
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
    command_joints:
      - joint_0
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

    state_publish_rate: 50.0 # Hz
    action_monitor_rate: 20.0 # Hz
    allow_partial_joints_goal: false
    allow_nonzero_velocity_at_trajectory_end: false
    constraints:
      stopped_velocity_tolerance: 0.02
      goal_time: 2.0

gripper_controller:
  ros__parameters:
    joint: left_carriage_joint
    allow_stalling: true

gravity_compensation_controller:
  ros__parameters:
    joints:
      - joint_0
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - left_carriage_joint

cartesian_pose_controller:
  ros__parameters:
    gpio_names:
      - ee_pos_x
      - ee_pos_y
      - ee_pos_z
      - ee_rot_x
      - ee_rot_y
      - ee_rot_z
